### DISCUSSION TITLE
skywalking CorrelationContext Correlation.ELEMENT_MAX_NUMBER

### DISCUSSION THREAD

**Discussion by zuomengjun** [apache/skywalking#8620] [Q&A] [answered]
The number of CorrelationContext elements in skywalking is limited to 3.
Does it matter if I increase the number of elements?

![image](https://user-images.githubusercontent.com/29054383/156721081-4b6b7a3f-b1e5-458b-9976-7f07f43fc61e.png)

**Comment by wu-sheng** [comment, marked as answer]
You could, which is why this is a configuration. If you set it too large, 
1. Your network proxy would cut the propagation, as it will propagate in your RPC and MQ calls. Proxy usually has a length limitation of HTTP header.
2. The memory will cost more as you hold more things in memory and tracing context.

3 is the same number, you could choose your own.

**Reply by zuomengjun** [reply]
I don't want the data to go over the network, can I clean it up before calling HTTP or MQ?

**Reply by wu-sheng** [reply]
Then why do you need `CorrelationContext`? This is the only purpose to design this.
